src/Microsoft.OpenApi.OData.Reader/OpenApiConvertSettings.cs

```diff
@@ -248,6 +248,11 @@ public string PathPrefix
         /// </summary>
         public Dictionary<string, string> CustomXMLAttributesMapping { get; set; } = new();
 
+        /// <summary>
+        /// Gets/sets a value indicating whether or not to append bound operations on derived types.
+        /// </summary>
+        public bool AppendBoundOperationsOnDerivedTypes { get; set; } = true;
```

Comment:
From what I understand this should actually be AppendBoundOperationsOnDerivedTypeCastSegments.

Now, I'm questioning the value of such a setting. The operations were added to the derived type segments when I implemented the type cast segments because I didn't give too many thoughts about that specific aspect.

The only value of adding these operations is to allow the caller to pass (or get as result) a more specialized derived type instead of the more general base type. But in most OOP languages this should already be possible thanks to auto upcasting (e.g. if I have an operation accepting a directoryObject, I should be able to pass it a User).

Besides increasing the number of path items and the size of SDKs I don't see any other value to this, and since it was only added in 1.0.10, I'd argue we remove that behavior all together, where if one wants to call an operation, they must call it on the non-type cast path item. Thoughts?

Reply:
Yes, the suggested name for the convert setting makes more sense.

From my tests, the particular path mentioned in the [issue](https://github.com/microsoft/OpenAPI.NET.OData/issues/221), `/servicePrincipals/{servicePrincipal-id}/createdObjects/{directoryObject-id}/microsoft.graph.servicePrincipal/microsoft.graph.addKey` gets its bound operation segment path generated due to this [condition](https://github.com/microsoft/OpenAPI.NET.OData/blob/0bfd9b35b8e9443669053aa4ff53f954f08d008f/src/Microsoft.OpenApi.OData.Reader/Edm/ODataPathProvider.cs#L671). So, even removing this [method](https://github.com/microsoft/OpenAPI.NET.OData/blob/0bfd9b35b8e9443669053aa4ff53f954f08d008f/src/Microsoft.OpenApi.OData.Reader/Edm/ODataPathProvider.cs#L807-L871) that explicitly adds the type cast and operation segments to a navigation property path will not resolve the issue highlighted above. There is also the possibility that a client outside MS Graph might have already taken a dependency on this particular version, hence removing the path might be a breaking change for them. Unless this kind of path is entirely not supported by OData. 

@peombwa any thoughts?

Reply:
if we want to keep the setting in the rare case of somebody having already taken a dependency on it in the last 3 weeks, then I'd suggest it's false by default.